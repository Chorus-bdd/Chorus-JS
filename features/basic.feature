Uses: StepServer
Uses: Selenium

Feature: Basic steps

  #! StepServer start
  Feature-Start:
    Given I open Chrome

  Background:
    Given I navigate to http://localhost:9000
    And StepServer client SimpleStepPublisher is connected

  Scenario: I can call steps with and without a result
    Check I can call a step with a result
    And I can call a step without a result

  Scenario: I can call steps which fail
    Check I can call a step with a result
    And I can call a step which fails

  Scenario: I can call steps which are async
    Check I can call a step which succeeds asychronously
    And I can call a step which times out
