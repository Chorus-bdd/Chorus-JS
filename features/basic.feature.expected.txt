Feature: Basic steps
  Scenario: Feature-Start
    #! StepRegistry start                                                                                        PASSED
    Given I open Chrome                                                                                          PASSED
  Scenario: I can call steps with and without a result
    Given I navigate to http://localhost:9999                                                                    PASSED
    And StepRegistry client SimpleStepPublisher is connected                                                     PASSED
    Check I can call a step with a result                                                                        PASSED  hello!
    And I can call a step without a result                                                                       PASSED
  Scenario: I can call steps which fail
    Given I navigate to http://localhost:9999                                                                    PASSED
    And StepRegistry client SimpleStepPublisher is connected                                                     PASSED
    Check I can call a step with a result                                                                        PASSED  hello!
    And I can call a step which fails                                                                            FAILED  (WebSocketClientStepInvoker:181)-StepFailedException Expected true to be false
  Scenario: I can call steps which are async
    Given I navigate to http://localhost:9999                                                                    PASSED
    And StepRegistry client SimpleStepPublisher is connected                                                     PASSED
    Check I can call a step which succeeds asynchronously                                                        PASSED  true
    And I can call a step which times out                                                                        FAILED  (WebSocketClientStepInvoker:181)-StepFailedException Expected false to be true
  Scenario: I can read a variable from the Chorus Context
    Given I navigate to http://localhost:9999                                                                    PASSED
    And StepRegistry client SimpleStepPublisher is connected                                                     PASSED
    Given I create a variable outbound with the value one                                                        PASSED
    Then in chorus-js 'outbound' has the value 'one'                                                             PASSED
  Scenario: I can set a variable in the Chorus Context
    Given I navigate to http://localhost:9999                                                                    PASSED
    And StepRegistry client SimpleStepPublisher is connected                                                     PASSED
    When I set the 'inbound' variable to 'two' in chorus-js                                                      PASSED
    Then the variable inbound has the value two                                                                  PASSED
  Scenario: I can overwrite a variable in the Chorus Context
    Given I navigate to http://localhost:9999                                                                    PASSED
    And StepRegistry client SimpleStepPublisher is connected                                                     PASSED
    Given I create a variable outbound with the value two                                                        PASSED
    When I set the 'outbound' variable to 'three' in chorus-js                                                   PASSED
    Then the variable outbound has the value three                                                               PASSED


Features  (total:1) (passed:0) (failed:1)
Scenarios (total:6) (passed:4) (failed:2)
Steps     (total:27) (passed:25) (failed:2) (undefined:0) (pending:0) (skipped:0)
